packages/material-ui/test/integration/Menu.test.js

```diff
@@ -1,403 +1,313 @@
-      } = {},
-    ) => {
-      const theme =
-        themeDirection !== undefined
-          ? createMuiTheme({
-              direction: themeDirection,
-            })
-          : undefined;
-
-      fill(contentAnchorTracker, false);
-      fill(focusTracker, false);
-      menuListFocusTracker = false;
-      fill(tabIndexTracker, false);
-      mount(
-        <Menu
-          variant={variant}
-          autoFocus={autoFocus}
-          anchorEl={document.body}
-          open
-          theme={theme}
-          MenuListProps={{
-            onFocus: () => {
-              menuListFocusTracker = true;
-            },
-          }}
-        >
-          {[0, 1, 2].map(itemIndex => {
-            if (itemIndex === invalidIndex) {
-              return null;
-            }
-            return (
-              <TrackingMenuItem
-                key={itemIndex}
-                disabled={itemIndex === disabledIndex ? true : undefined}
-                itemIndex={itemIndex}
-                selected={itemIndex === selectedIndex}
-                tabIndex={itemIndex === selectedIndex ? selectedTabIndex : undefined}
-                autoFocus={itemIndex === autoFocusIndex ? true : undefined}
-              >
-                Menu Item {itemIndex}
-              </TrackingMenuItem>
-            );
-          })}
-        </Menu>,
+
+    it('[variant=selectedMenu] nothing selected', () => {
+      const { getAllByRole, getByRole } = render(
+        <OpenMenu variant="selectedMenu">
+          <MenuItem />
+          <MenuItem />
+          <MenuItem />
+        </OpenMenu>,
       );
-    };
-
-    it('[variant=menu] adds coverage for rtl and Tab with no onClose', () => {
-      // This isn't adding very meaningful coverage apart from verifying it doesn't error, but
-      // it was so close to 100% that this has value in avoiding needing to drill into coverage
-      // details to see what isn't being tested.
-      mountTrackingMenu('menu', { themeDirection: 'rtl' });
-      assert.deepEqual(contentAnchorTracker, [true, false, false]);
-      assert.deepEqual(focusTracker, [false, false, false]);
-      assert.strictEqual(menuListFocusTracker, true);
-      assert.deepEqual(tabIndexTracker, [false, false, false]);
-
-      // Adds coverage for Tab with no onClose
-      TestUtils.Simulate.keyDown(document.activeElement, {
-        key: 'Tab',
-      });
-    });
 
-    it('[variant=menu] nothing selected', () => {
-      assert.deepEqual(contentAnchorTracker, [true, false, false]);
-      assert.deepEqual(focusTracker, [false, false, false]);
-      assert.strictEqual(menuListFocusTracker, true);
-      assert.deepEqual(tabIndexTracker, [false, false, false]);
+      expect(getByRole('menu')).to.be.focused;
+      getAllByRole('menuitem').forEach(item => expect(item).to.have.property('tabIndex', -1));
     });
 
+    // no case for variant=selectedMenu
     it('[variant=menu] nothing selected, autoFocus on third', () => {
```

Comment:
This is currently testing that the focus on a menuitem is overwritten by the focus on the menu. Is this actually expected? 

One potentially confusing part is that `autoFocus` is by default true on `Menu` but false on `MenuItem`. Apart from violating API naming convention it can be confusing as well. If I see `<Parent><Child autoFocus /></Parent>` I expect the child to be focused not the parent.

I would much rather warn to avoid confusion or potential race conditions.

Reply:
> This is currently testing that the focus on a menuitem is overwritten by the focus on the menu.

The `menuListFocusTracker` variable is a bit confusing. Since this is set by `onFocus` which bubbles up, this is always true whether the list has focus or an item (only false if neither gets focus). It only means that the list has focus if this is true and none of the item focus trackers are true. If focus events were actually fired on both the list and an item, the test isn't written in a way that can differentiate that from just the item having a focus event.

Reply:
> If focus events were actually fired on both the list and an item, the test isn't written in a way that can differentiate that from just the item having a focus event.

So do you think it is important to test that both the whole list and a specific item got focused? Shouldn't this be considered a defect (fast focus+blur) that we want to avoid?

Reply:
> Shouldn't this be considered a defect

I think it’s just a bad test. It would be more valid if menu autoFocus was false. Explicit autoFocus on the items means the user is managing the focus. I didn’t make any attempt to detect use of item autoFocus and automatically turn off menu autoFocus in that case.

Reply:
I mean it is good to test how the component behaves in those inconsistent states. Whether it makes sense to apply this props combination is another discussion but it is currently possible so we need to document it.

So either way the test is important. Either to document a bug or behavior of non-sensical props combinations (since we want to iterate on the API to not allow those).